src/wasm/wasm-emscripten.cpp

```diff
@@ -769,7 +775,8 @@ std::string EmscriptenGlueGenerator::generateEmscriptenMetadata(
   commaFirst = true;
   for (const auto& import : wasm.imports) {
     if (import->base.startsWith("invoke_")) {
-      meta << maybeComma() << '"' << import->base.str << '"';
+      if (importedNames.insert(import->base.str).second)
```

Comment:
Kind of weird that we reuse importedNames for both declares and invokeFuncs. This is correct, because we have the invariant that invokeFuncs start with "invoke_", and declares explicitly do not, but there's intervening code that makes that non-obvious.

I'd say use two std::sets to make it obvious that they don't conflict

Reply:
Done